packages/openapi-generator/src/parser/media-type.ts

```diff
@@ -0,0 +1,85 @@
+import { createLogger } from '@sap-cloud-sdk/util';
+import { OpenAPIV3 } from 'openapi-types';
+import { OpenApiSchema } from '../openapi-types';
+import { parseSchema } from './schema';
+
+const logger = createLogger('openapi-generator');
+/**
+ * Parse the type of a resolved request body or response object.
+ * @param bodyOrResponseObject The request body or response object to parse the type from.
+ * @returns The type name of the request body if there is one.
+ */
+export function parseApplicationJsonMediaType(
+  bodyOrResponseObject:
+    | OpenAPIV3.RequestBodyObject
+    | OpenAPIV3.ResponseObject
+    | undefined
+): OpenApiSchema | undefined {
+  if (bodyOrResponseObject) {
+    const mediaType = getMediaTypeObject(
+      bodyOrResponseObject,
+      'application/json'
+    );
+    const schema = mediaType?.schema;
+    if (schema) {
+      return parseSchema(schema);
+    }
+  }
+}
+
+export function parseMediaType(
+  bodyOrResponseObject:
+    | OpenAPIV3.RequestBodyObject
+    | OpenAPIV3.ResponseObject
+    | undefined
+): OpenApiSchema | undefined {
+  const allMediaTypes = getMediaTypes(bodyOrResponseObject);
+  if (allMediaTypes.length) {
+    const jsonMediaType = parseApplicationJsonMediaType(bodyOrResponseObject);
+
+    if (!jsonMediaType) {
+      logger.debug(
+        "Could not parse media type, because it is not 'application/json'. Generation will continue with 'any'. This might lead to errors at runtime."
```

Comment:
Should we give here a hint that if you want to make a call and it is really not application/json use `executeRaw`? Or should we add this warning at runtime? Before the crash will most likely appear.

Reply:
Actually they can also use execute, we don't parse anything, we just return the response.data. I would rather see that in the API doc of the function. because they have to set the header themselves.